File: game-api/src/main/java/org/runestar/client/game/api/GameState.java
package org.runestar.client.game.api;

public final class GameState {

    private GameState() {}

    public static final int NONE = 0;
    public static final int STARTING = 5;
    public static final int TITLE = 10;
    public static final int AUTHENTICATOR = 11;
    public static final int LOGGING_IN = 20;
    public static final int LOADING = 25;
    public static final int LOGGED_IN = 30;
    public static final int CONNECTION_LOST = 40;
    public static final int CHANGING_WORLD = 45;
}

File: game-api/src/main/java/com/runesuite/client/game/api/GameState.kt
package com.runesuite.client.game.api

enum class GameState(val id: Int) {

    NONE(0),
    STARTING(5),
    TITLE(10),
    // 11 ?
    LOGGING_IN(20),
    LOADING(25),
    LOGGED_IN(30),
    CONNECTION_LOST(40),
    CHANGING_WORLD(45);

    companion object {

        @JvmField
        val LOOKUP = values().associateBy { it.id }
    }
}